sweepbatcher/sweep_batcher_test.go

```diff
@@ -710,26 +739,33 @@ func testSweepBatcherSimpleLifecycle(t *testing.T, store testStore,
 	// Deliver sweep request to batcher.
 	require.NoError(t, batcher.AddSweep(&sweepReq1))
 
-	// Eventually request will be consumed and a new batch will spin up.
-	require.Eventually(t, func() bool {
-		return len(batcher.batches) == 1
-	}, test.Timeout, eventuallyCheckFrequency)
-
 	// When batch is successfully created it will execute it's first step,
 	// which leads to a spend monitor of the primary sweep.
 	<-lnd.RegisterSpendChannel
 
+	// Eventually request will be consumed and a new batch will spin up.
+	require.Eventually(t, func() bool {
+		return len(getBatches(ctx, batcher)) == 1
```

Comment:
Alternatively we could add a `Batcher.NumBatches()` and a `batch.NumSweeps()` so most of these repeating checks are easier to write.

Reply:
Perhaps we could also add a `batch.Sweeps()` which would return the swap hashes for the sweeps. So then we could completely remove the intrusive check.

Reply:
I added private methods `Batcher.numBatches(ctx)`, `batch.numSweeps(ctx)` and `batch.snapshot(ctx)` and managed to get rid of the majority of `testRunInEventLoop` calls.